presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java

```diff
@@ -298,8 +301,14 @@ private static ConnectorPageSource createParquetPageSource(
             HiveFileContext hiveFileContext = new HiveFileContext(true, NO_CACHE_CONSTRAINTS,
                     Optional.empty(), Optional.of(fileSize), modificationTime, false);
             FSDataInputStream inputStream = fileSystem.openFile(path, hiveFileContext);
-            dataSource = buildHdfsParquetDataSource(inputStream, path, fileFormatDataSourceStats);
-            ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, fileSize).getParquetMetadata();
+            final ParquetDataSource parquetDataSource = buildHdfsParquetDataSource(inputStream, path, fileFormatDataSourceStats);
+            dataSource = parquetDataSource;
```

Comment:
why are there both parquetDataSource and dataSource? is there any particular meaning?

Reply:
It is mainly because the lampda expression below (line 311). If I use datasource in that line, then there is build error "Variable used in lambda expression should be final or effectively final".

Reply:
I see, that make sense, thx!